src/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java

```diff
@@ -880,8 +880,10 @@ public boolean _execute() throws Exception {
       }
     }
     // SP-307, BISERVER-9688
-
     final MasterReport report = getReport();
+    // Set unlimited queryLimit for reports running in "Run in background mode" in PUC.
+    report.setQueryLimit( -1 );
```

Comment:
Never *sets* a row-limit. But that does not mean that we remove a row-limit that has been set by the user previously. This attribute is editable in the report-designer and therefore a user can set their own limits here. 

So in here you leave the query limit alone - if the user has defined (in PRD) that their report shall never return more than 10 rows, then this is what the user shall get.